src/pointtopoint.jl

```diff
@@ -411,6 +411,19 @@ function Wait!(req::Request)
     stat_ref[]
 end
 
+function Base.wait(req::Request)
+    stat_ref = Ref{Status}(MPI.STATUS_EMPTY)
+    alreadynull = isnull(req)
+    errcode = @threadcall((:MPI_Wait, libmpi), Cint,
+                          (Ptr{MPI_Request}, Ptr{Status}),
+                          req, stat_ref)
```

Comment:
One issue is that on 32-bit windows we need to use the `stdcall` convention: this is usually handled by the `@mpicall`/`@mpichk` macros, but `@threadcall` doesn't seem to support nonstandard calling conventions.

On the other hand, it passed tests so 🤷 ?

Reply:
(alternatively, we could disable this on 32-bit Windows, like we do the custom operator stuff)